### ISSUE TITLE
l10n_it_edi: no preview for electronic invoice attachments

### ISSUE THREAD

**Issue by odooNextev** [odoo/odoo#153589]
Impacted versions:
- [ ] 12.0
- [ ] 13.0
- [ ] 14.0
- [ ] 15.0
- [ ] 16.0
- [ ] 17.0

Steps to reproduce:

1. Install `l10n_it_edi` in a runbot and switch to `IT Company` which was automatically created
2. Go to Invoicing -> Customers -> Invoices
3. Open an existing one or create a new one, post it and generate `E-invoice XML` (click `send & print` and check `E-invoice XML`)
4. Go to Invoicing -> Vendors -> Bills
5. Upload an XML invoice file through `Upload` button

Current behavior:
XML files are treated like generic ir.attachment and you can only download them

Expected behavior:
It would be useful to allow users to show XML files preview in AdE and AssoInvoice format as Italian Localization did in  `l10n_it_fatturapa`.
It's very important, in particular for vendor bills, to check between the original one and what Odoo has elaborated.
Without this feature users have to download files and open in an external software to render them in an understandable way.

Can I try to do a PR or do you think this feature is irrelevant?

@eLBati @lordkrandel

**Comment by lordkrandel** [comment]
@tsb-odoo ☝🏻 Feedback for you.

**Comment by salvorapi** [comment]
up

**Comment by dcorio** [comment]
Agree, without beeing able to print outs the content of the XML file using one of the two PDF layouts commonly used in Italy, one couldn't spot import errors, missing data, etc...
It a must-have feature since we cannot ask users to download the XML and use a 3rd-party software to generate the PDF version.

**Comment by odooNextev** [comment]
@tsb-odoo thank you for your reply.
As soon as I can I'll link a PR to this issue, then you can have a look at it and judge whether it is already good like this or it will be better to start from scratch.

**Comment by eLBati** [comment]
Here is the PR https://github.com/odoo/odoo/pull/160712 thanks to @borruso
@odooNextev could you add it to description?
